src/date.cpp

```diff
@@ -199,8 +199,8 @@ static void OnNewYear()
 
 	if (_cur_year == _settings_client.gui.semaphore_build_before) ResetSignalVariant();
 
-	/* check if we reached end of the game */
-	if (_cur_year == ORIGINAL_END_YEAR) {
+	/* check if we reached end of the game (end of ending year) */
+	if (_cur_year == _settings_game.game_creation.ending_year + 1) {
```

Comment:
In the original TTD, the game shows the score screen between 31st december 2049 and 1st january 2050. OpenTTD defaults adds +1 year. Since you're editing this part of the game, perhaps you could make it behave like TTD?